src/SQLQueryStress/DatabaseSelect.cs

```diff
@@ -84,75 +84,146 @@ public DatabaseSelect(Form1.QueryStressSettings settings)
             authentication_comboBox.SelectedIndexChanged += authentication_comboBox_SelectedIndexChanged;
             pm_authentication_comboBox.SelectedIndexChanged += pm_authentication_comboBox_SelectedIndexChanged;
 
-            db_comboBox.Click += db_comboBox_Click;
-            pm_db_comboBox.Click += pm_db_comboBox_Click;
+            db_comboBox.Enter += Db_comboBox_Enter;
+            db_comboBox.Leave += Db_comboBox_Leave;
+
+            pm_db_comboBox.Enter += Db_comboBox_Enter;
+            pm_db_comboBox.Leave += Db_comboBox_Leave;
+            
+            server_textBox.KeyDown += Server_textBox_KeyDown;
+            server_textBox.TextChanged += Server_textBox_TextChanged;
+
+            Server_textBox_TextChanged(null, null);
         }
 
-        private void authentication_comboBox_SelectedIndexChanged(object sender, EventArgs e)
+        private void Server_textBox_TextChanged(object sender, EventArgs e)
         {
-            if (authentication_comboBox.SelectedIndex == 0)
+            if (server_textBox.Text == string.Empty)
             {
-                login_textBox.Enabled = false;
-                password_textBox.Enabled = false;
+                ok_button.Enabled = false;
             }
             else
             {
-                login_textBox.Enabled = true;
-                password_textBox.Enabled = true;
+                ok_button.Enabled = true;
             }
         }
 
-        private void cancel_button_Click(object sender, EventArgs e)
+        private void Server_textBox_KeyDown(object sender, KeyEventArgs e)
         {
-            Dispose();
+            db_comboBox.SelectedIndex = -1;
         }
 
-        private void db_comboBox_Click(object sender, EventArgs e)
+        private void Db_comboBox_Leave(object sender, EventArgs e)
         {
-            SaveLocalSettings();
-
-            var selectedItem = (string) db_comboBox.SelectedItem;
+            ComboBox cbSender = ((ComboBox)sender);
+            if ((cbSender.SelectedValue == null || cbSender.Text != cbSender.SelectedItem.ToString()) && cbSender.Items.Contains(cbSender.Text))
+            {
+                cbSender.SelectedItem = cbSender.Text;
+            }
+        }
 
-            var sql = "" + "SELECT name " + "FROM master..sysdatabases " + "ORDER BY name";

```

Comment:
I'm not sure why this line was fragmented. I suspect a previous developer may have tried to keep each clause of this SELECT statement on separate lines. I've tidied this and added a state filter to a commit for which I'll submit another pull request in a moment.